manifests/params.pp

```diff
@@ -19,7 +19,7 @@
       $service_provider = 'upstart'
     }
     'RedHat': {
-      if ($::operatingsystemmajrelease == '6') {
+      if ($::operatingsystemmajrelease == '6' or $::operatingsystem =~ /^[Aa]mazon$/) {
```

Comment:
Something about this regex is causing the tests to fail.
Most other puppet modules just match against the string `Amazon` , so I'm happy to relax this regex and just do a plain string comparison.